File: gracie.lee/src/main/java/homework/chapter_6/homework3/Member.java
package homework.chapter_6.homework3;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "Hw3Member")
@Table(name = "member_hw3")
public class Member {

    @Id
    @GeneratedValue
    @Column(name = "member_id")
    private Long id;

    private String name;

    @OneToMany(mappedBy = "member")  // ← Order.member 필드를 가리킴 (읽기 전용)
    private List<Order> orders = new ArrayList<>();

    protected Member() {
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<Order> getOrders() {
        return orders;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: gracie.lee/src/main/kotlin/homework/chapter_6/homework3/Member.kt
package homework.chapter_6.homework3

import jakarta.persistence.*

@Entity(name = "Hw3MemberKt")
@Table(name = "member_hw3_kt")
class Member(
    @Id
    @GeneratedValue
    @Column(name = "member_id")
    val id: Long? = null,

    var name: String? = null,

    @OneToMany(mappedBy = "member")  // <- Order.member 필드를 가리킴 (읽기 전용)
    val orders: MutableList<Order> = mutableListOf()
)
